perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ModulesUtilsBlImpl.java

```diff
@@ -481,14 +486,40 @@ public boolean haveRightToWriteAttributeInAnyGroupOrResource(PerunSessionImpl se
 		return groupNameNamespaces;
 	}
 
-	public void checkReservedUnixGroupNames(Attribute groupNameAttribute) throws WrongAttributeValueException {
+	public void checkReservedUnixGroupNames(Attribute groupNameAttribute) throws InternalErrorException, WrongAttributeValueException {
 		if(groupNameAttribute == null) return;
-		if(reservedNamesForUnixGroups.contains(groupNameAttribute.getValue())) throw new WrongAttributeValueException(groupNameAttribute, "This groupName is reserved.");
+
+		if (perunNamespaces == null) {
+			perunNamespaces = BeansUtils.getAllPropertiesFromCustomConfiguration("perun-namespaces.properties");
+		}
+		String reservedNames;
+		if ((reservedNames = perunNamespaces.get(groupNameAttribute.getFriendlyName() + ":reservedNames")) != null) {
+			List<String> reservedNamesList = Arrays.asList(reservedNames.split("\\s*,\\s*"));
+			if (reservedNamesList.contains(groupNameAttribute.getValue()))
+				throw new WrongAttributeValueException(groupNameAttribute, "This groupName is reserved.");
+		} else {
+			//Property not found in our attribute map, so we use the default hardcoded values instead
+			if (reservedNamesForUnixGroups.contains(groupNameAttribute.getValue()))
+				throw new WrongAttributeValueException(groupNameAttribute, "This groupName is reserved.");
+		}
 	}
 
-	public void checkUnpermittedUserLogins(Attribute loginAttribute) throws WrongAttributeValueException {
+	public void checkUnpermittedUserLogins(Attribute loginAttribute) throws InternalErrorException, WrongAttributeValueException {
 		if(loginAttribute == null) return;
-		if(unpermittedNamesForUserLogins.contains(loginAttribute.getValue())) throw new WrongAttributeValueException(loginAttribute, "This login is not permitted.");
+
+		if (perunNamespaces == null) {
+			perunNamespaces = BeansUtils.getAllPropertiesFromCustomConfiguration("perun-namespaces.properties");
+		}
+		String unpermittedNames;
+		if ((unpermittedNames = perunNamespaces.get(loginAttribute.getFriendlyName() + ":reservedNames")) != null) {
```

Comment:
For better readability please fill it when declare variable and in the if clause just test if it is not null.